### PR TITLE
Add prettier and husky hook

### PR DESCRIPTION
**Pull request by tomdye**
**Type:** feature

The following has been addressed in the PR:

* [ ] There is a related issue
* [x] All code matches the [style guide](https://github.com/dojo/framework/blob/master/STYLE.md)
* [ ] Unit or Functional tests are included in the PR

**Description:**

- Add's `prettier` and husky pre commit hook.
- initial prettier run
- future widgets to be written in `tsx`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -46,9 +52,25 @@
...
+    "typescript": "~2.6.1",
+    "husky": "2.2.0",
+    "lint-staged": "8.1.6",
+    "prettier": "1.17.0"
   },
   "dependencies": {
     "tslib": "~1.8.1"
+  },
+  "lint-staged": {
+    "*.{ts,tsx}": [
```
Comment (aciccarello) [inline package.json]: should also lint css

Comment (tomdye) [reply package.json]: would require a different linter etc though right?

**Comment by tomdye** [comment]
Only the `package.json` has been manually updated.

File: package.json (hunk 1 above)
Comment (aciccarello) [reply package.json]: I think it picks it up automatically

**Comment by msssk** [comment]
Do not like prettier. It completely fails to live up to its name.

**Comment by sammenza** [comment]
Late to the party but this looks great! Super useful, thanks for setting up 🎉

**Pull request merged**